metadata-ingestion/src/datahub/ingestion/source/gc/soft_deleted_entity_cleanup.py

```diff
@@ -147,41 +154,50 @@ def cleanup_soft_deleted_entities(self) -> None:
         assert self.ctx.graph
 
         deleted_count_retention = 0
-        urns = self.ctx.graph.get_urns_by_filter(
-            entity_types=self.config.entity_types,
-            platform=self.config.platform,
-            env=self.config.env,
-            query=self.config.query,
-            status=RemovedStatusFilter.ONLY_SOFT_DELETED,
-            batch_size=self.config.batch_size,
-        )
+        while (
+            self.report.num_soft_deleted_entity_removed
+            <= self.config.limit_entities_delete
+        ):
+            urns = list(
```

Comment:
You get here more items than the limit. 
I think it would be better to only iterate over the `get_urns_by_filter` and stop when you hit the limit and only keep `limit` number of item in memory and in the list

Reply:
Do you mean in the case where batch_size > limit?

Reply:
What if I do this
```
         while (
-            self.report.num_soft_deleted_entity_removed
-            <= self.config.limit_entities_delete
+            self.config.limit_entities_delete <= 0
+            or self.report.num_soft_deleted_entity_removed
+            <= max(self.config.limit_entities_delete, self.config.batch_size)
         ):

```